src/main/java/com/uwetrottmann/tmdb2/TmdbInterceptor.java

```diff
@@ -62,17 +62,18 @@ public static Response handleIntercept(Chain chain, Tmdb tmdb) throws IOExceptio
             // re-try if the server indicates we should
             String retryHeader = response.header("Retry-After");
             if (retryHeader != null) {
+                // close body of unsuccessful response
+                response.close();
+
                 try {
                     int retry = Integer.parseInt(retryHeader);
                     Thread.sleep((int) ((retry + 0.5) * 1000));
 
-                    // close body of unsuccessful response
-                    if (response.body() != null) {
-                        response.body().close();
-                    }
```

Comment:
Closing the response only here because `Thread.sleep()` might throw. And in that case the original response should be returned. And any returned response must not be closed. Not sure how else to solve this.